Entities/AtomGroup.cpp

```diff
@@ -1,2618 +1,1750 @@
+			}
+		}
+
+		// Now actually apply the exit vectors to both, but only if the jump isn't too jarring
+		if (thisExit.GetMagnitude() < m_OwnerMO->GetRadius()) { position += thisExit; }
+		if (!intersectedExit.IsZero() && intersectedExit.GetMagnitude() < intersectedMO->GetRadius()) { intersectedMO->SetPos(intersectedMO->GetPos() + intersectedExit); }
+
+		// If we've been pushed into the terrain enough, just gib as the squashing effect
+		if (m_OwnerMO->CanBeSquished() && RatioInTerrain() > 0.75F) /* && totalExitVector.GetMagnitude() > m_OwnerMO->GetDiameter()) */ {
+			// Move back before gibbing so gibs don't end up inside terrain
+			position -= thisExit;
+			m_OwnerMO->GibThis(-totalExitVector);
+		}
+
+		MOSRotating *intersectedMOS = dynamic_cast<MOSRotating *>(intersectedMO);
+
+		if (intersectedMOS && intersectedMOS->CanBeSquished() && intersectedMOS->GetAtomGroup()->RatioInTerrain() > 0.75F) /* && totalExitVector.GetMagnitude() > intersectedMO->GetDiameter()) */ {
+			// Move back before gibbing so gibs don't end up inside terrain
+			intersectedMO->SetPos(intersectedMO->GetPos() - intersectedExit);
+			intersectedMOS->GibThis(totalExitVector);
+		}
+
+		// TODO: this isn't really true since we don't check for clearness after moving the position
+		return intersectingAtoms.empty();
 	}
 
-    // Restart and go through all atoms to find all intersecting the specific intersected MO
-    for (Atom *atom : m_Atoms)
-    {
-        atomPos = atom->GetCurrentPos();
-        if (g_SceneMan.GetMOIDPixel(atomPos.GetFloorIntX(), atomPos.GetFloorIntY()) == currentMOID)
-        {
-            // Add atom to list of intersecting ones
-            intersectingAtoms.push_back(atom);
-        }
-    }
-
-    // No intersections - we're clear?!
-//        if (intersectingAtoms.empty())
-//            return true;
-//        RTEAssert(!intersectingAtoms.empty(), "Couldn't find any intersections after finding one?!");
-
-    // If all atoms are intersecting, we're screwed?!
-//        if (intersectingAtoms.size() >= m_Atoms.size())
-//            return false;
-
-    // Go through all intesecting atoms and find their average inverse normal
-	for (const Atom *intersectingAtom : intersectingAtoms) {
-		exitDirection += m_pOwnerMO->RotateOffset(intersectingAtom->GetNormal());
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void AtomGroup::Draw(BITMAP *targetBitmap, const Vector &targetPos, bool useLimbPos, unsigned char color) const {
+		Vector atomPos;
+		Vector normal;
+
+		acquire_bitmap(targetBitmap);
+
+		for (const Atom *atom : m_Atoms) {
+			if (!useLimbPos) {
+				atomPos = (m_OwnerMO->GetPos() + (atom->GetOffset().GetXFlipped(m_OwnerMO->m_HFlipped) * m_OwnerMO->GetRotMatrix()));
+			} else {
+				atomPos = (m_LimbPos + (atom->GetOffset().GetXFlipped(m_OwnerMO->m_HFlipped) * m_OwnerMO->GetRotMatrix()));
+			}
+			// Draw normal first, then draw the Atom position
```

Comment:
I don't think we really need this comment, just describes the code. If you wanted to change it to explain what drawing the normal means or why we want to do it then I'm in favour, but as it is, it's better off dead.